src/test/java/be/seeseemelk/mockbukkit/block/data/WallSignMockTest.java

```diff
@@ -0,0 +1,141 @@
+package be.seeseemelk.mockbukkit.block.data;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.IOException;
+import java.util.Set;
+
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+
+import be.seeseemelk.mockbukkit.MockBukkit;
+import be.seeseemelk.mockbukkit.ServerMock;
+import be.seeseemelk.mockbukkit.WorldMock;
+
+class WallSignMockTest {
+
+	private WallSignMock sign;
+	private @NotNull ServerMock server;
+
+	@BeforeEach
+	void setUp() throws IOException {
+		MockBukkit.mock();
+		sign = new WallSignMock(Material.ACACIA_WALL_SIGN);
+	}
+
+	@AfterEach
+	void tearDown() {
+		MockBukkit.unmock();
+	}
+
+	@Test
+	@Order(1)
+	void constructor_FullConstruction() {
+		WorldMock world = new WorldMock();
+		Block signBlock = world.getBlockAt(new Location(world, 0, 2, 1));
+		signBlock.setType(Material.OAK_WALL_SIGN);
+
+		Assertions.assertEquals(WallSignMock.class, signBlock.getBlockData().getClass());
+	}
```

Comment:
This test is redundant because the `blockDataMock_Mock_CorrectType` tests that `BlockDataMock#mock` returns the correct type and `BlockMockTest#setType_SetToMaterialWithBlockDataMockSubclass` ensures the proper block data is set from the setType method.